src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,260 @@
+  for (size_t i = 0; i< data.n_cols; i++)
+  {
+    if (labels(i) == 0)
+      label(i) = -1;
+    else
+      label(i) = 1;
+  }
+
+  alpha = arma::zeros(data.n_cols);
+  size_t count = 0;
+  arma::vec E = arma::zeros(data.n_cols, 1);
+  double eta = 0;
+  double L = 0;
+  double H = 0;
+  arma::mat K = arma::mat(data.n_cols, data.n_cols);
+
+  for (size_t i = 0; i < data.n_cols; i++)
+  {
+    for (size_t j = 0; j < data.n_cols; j++)
+    {
+      K(i, j) = kernel.Evaluate(data.col(i), data.col(j));
+    }
+  }
+  while (count < max_iter)
+  {
+    size_t num_changed_alphas = 0;
+    for (size_t i = 0; i < data.n_cols; i++)
+    {
+      E(i) = b + arma::sum(alpha.t() % label % K.col(i).t()) - label(i);
+      if ((label(i) * E(i) < -tol && alpha(i) < C) ||
+        (label(i) * E(i) > tol && alpha(i) > 0))
+      {
+        size_t j = rand() % data.n_cols;
+        while (j == i)
+        {
+          j = rand() % data.n_cols;
+        }
+        E(j) = b + arma::sum(alpha.t() % label % K.col(j).t()) - label(j);
+
+        double alpha_i_old = alpha(i);
+        double alpha_j_old = alpha(j);
+
+        if (label(i) == label(j))
+        {
+          L = std::max(0.0, alpha(j) + alpha(i) - C);
+          H = std::min(C, alpha(j) + alpha(i));
+        }
+        else
+        {
+          L = std::max(0.0, alpha(j) - alpha(i));
+          H = std::min(C, C + alpha(j) - alpha(i));
+        }
+
+        if (L == H)
+          continue;
+
+        eta = 2 * K(i, j) - K(i, i) - K(j, j);
+        if (eta >= 0)
+          continue;
+
+        alpha(j) = alpha(j) - (label(j) * (E(i) - E(j))) / eta;
+
+        alpha(j) = std::min(H, alpha(j));
+        alpha(j) = std::max(L, alpha(j));
+
+        if (std::abs(alpha(j) - alpha_j_old) < tol)
+        {
+          alpha(j) = alpha_j_old;
+          continue;
+        }
+
+        alpha(i) = alpha(i) + label(i) * label(j) * (alpha_j_old - alpha(j));
+
+        double b1 = b - E(i)
+                    - label(i) * (alpha(i) - alpha_i_old) *  K(i, j)
+                    - label(j) * (alpha(j) - alpha_j_old) *  K(i, j);
+        double b2 = b - E(j)
+                    - label(i) * (alpha(i) - alpha_i_old) *  K(i, j)
+                    - label(j) * (alpha(j) - alpha_j_old) *  K(j, j);
+
+        if (0 < alpha(i) && alpha(i) < C)
+          b = b1;
+        else if (0 < alpha(j) && alpha(j) < C)
+          b = b2;
+        else
+          b = (b1 + b2) / 2;
+
+        num_changed_alphas = num_changed_alphas + 1;
+      }
+    }
+
+    if (num_changed_alphas == 0)
+      count = count + 1;
+    else
+      count = 0;
+  }
+  parameters = (data * (alpha.t() % label).t()).t();
+  double threshold = arma::as_scalar(arma::mean(alpha));
+
+  size_t number = 0;
+  y = arma::zeros(1, data.n_cols);
+  x = arma::zeros(data.n_rows, data.n_cols);
+  for (size_t i = 0; i < data.n_cols; i++)
+  {
+    if (alpha(i) > threshold)
+    {
+      y(i) = label(i);
+      x.col(i) = data.col(i);
```

Comment:
Hey @rcurtin memory builds are failing because of this implementation. `x.col(i)` is pointing to `data.col(i)`. When execution of train function is complete then `data` is deleted from memory and `x.col(i)` is still pointing to `data.col(i)`. Can you suggest any better way to initialize this?

Reply:
can you use a more descriptive name than `x` ? 
also you can create a copy of the data and try to use that instead?

Reply:
Sure will ad a description also will try to make copy but it will decrease the efficiency